### DISCUSSION TITLE
Q&A: Phase 38.5 FederatedOrchestrator — Pipeline Configuration & Integration

### DISCUSSION THREAD

**Discussion by web3guru888** [web3guru888/asi-build#786] [Q&A]
### Configuration Questions

**Q1: Centralized vs Decentralized vs Hierarchical — when to use each?**

| Mode | Topology | Best For | Limitations |
|------|----------|----------|-------------|
| Centralized | Star | Standard FL, ≤1000 clients | Single point of failure |
| Decentralized | Mesh/Ring | No trusted server, peer-to-peer | Slower convergence |
| Hierarchical | Tree | Edge computing, multi-tier | Complex coordination |

**Q2: How to balance privacy and accuracy?**
- Run multiple experiments with different ε values: [1, 2, 4, 8, 16]
- Use `get_pareto_frontier()` to find optimal configurations
- Rule of thumb: ε=8 gives <2% accuracy loss on CIFAR-10 with FedAvg
- Combine DP with secure aggregation for defense in depth

**Q3: How to detect convergence in federated settings?**
- Loss plateau: `patience=10` rounds without ≥0.1% improvement
- Accuracy target: stop when global accuracy ≥ target
- Privacy budget: stop when remaining ε < minimum for one round
- Gradient norm: stop when ‖Δw‖ < threshold (indicates minimum reached)

**Q4: What's the optimal number of clients per round?**
- McMahan et al. recommend C=0.1 (10% of clients per round)
- More clients per round = less noise, fewer rounds needed
- Fewer clients per round = better privacy amplification
- Sweet spot: 10-100 clients per round for 1000-client pool

**Q5: How does this integrate with existing ASI-Build phases?**
- **Phase 37** SwarmTopologyManager: provides communication topologies for decentralized FL
- **Phase 36** EvolutionaryOrchestrator: can evolve FL hyperparameters (aggregation strategy, local epochs)
- **Phase 33** ContinualOrchestrator: federated continual learning — prevent catastrophic forgetting across rounds
- **Phase 32** CollectiveOrchestrator: multi-agent FL coordination in hierarchical settings

**Q6: End-to-end example configuration?**
```python
config = FederatedConfig(
    mode=FederatedMode.CENTRALIZED,
    num_rounds=100,
    clients_per_round=10,
    local_epochs=5,
    local_batch_size=32,
    local_learning_rate=0.01,
    aggregation_strategy="fedavg",
    enable_dp=True,
    enable_secure_agg=False,
    convergence=ConvergenceCriterion.LOSS_PLATEAU,
    patience=10,
    target_accuracy=0.95
)
result = orchestrator.run_experiment(config)
```

See issue #776 for full specification and acceptance criteria.
